bigtable-dataflow-parent/bigtable-dataflow-import/src/main/java/com/google/cloud/bigtable/dataflowimport/HBaseExportJob.java

```diff
@@ -0,0 +1,303 @@
+  }
+
+  /**
+   * Reads each shard and writes its elements into a SequenceFile with compressed blocks.
+   */
+  static class WriteResultsToSeq extends DoFn<KV<String, BoundedSource<Result>>, Void> {
+    private final String basePath;
+    private final Aggregator<Long, Long> itemCounter;
+    private final BatchCounter readDuration;
+    private final BatchCounter writeDuration;
+    private final BatchCounter totalDuration;
+    private final BatchCounter bytesWritten;
+
+    private transient Configuration hadoopConfig;
+
+    WriteResultsToSeq(String basePath) {
+      this.basePath = basePath;
+
+      itemCounter = createAggregator("itemsProcessed", new SumLongFn());
+      readDuration = new BatchCounter(createAggregator("readDuration(secs)", new SumLongFn()), 1_000_000_000);
+      writeDuration = new BatchCounter(createAggregator("writeDuration(secs)", new SumLongFn()), 1_000_000_000);
+      totalDuration = new BatchCounter(createAggregator("totalDuration(secs)", new SumLongFn()), 1_000_000_000);
+      bytesWritten = new BatchCounter(createAggregator("bytesWritten(MB)", new SumLongFn()), 1024 * 1024);
+    }
+
+    @Override
+    public void populateDisplayData(DisplayData.Builder builder) {
+      super.populateDisplayData(builder);
+    }
+
+    @Override
+    public void startBundle(Context c) throws Exception {
+      hadoopConfig = new Configuration(false);
+
+      // Not using setStrings to avoid loading StringUtils which tries to access HADOOP_HOME
+      hadoopConfig.set("io.serializations", Joiner.on(',').join(
+          WritableSerialization.class.getName(), ResultSerialization.class.getName())
+      );
+      hadoopConfig.set("fs.gs.project.id", c.getPipelineOptions().as(DataflowPipelineOptions.class).getProject());
+      for (Map.Entry<String, String> entry : HBaseImportIO.CONST_FILE_READER_PROPERTIES.entrySet()) {
+        hadoopConfig.set(entry.getKey(), entry.getValue());
+      }
+
+      super.startBundle(c);
+    }
+
+    @Override
+    public void processElement(ProcessContext c) throws Exception {
+      final String shardId = c.element().getKey();
+      final BoundedSource<Result> source = c.element().getValue();
+
+      LOG.info("Starting new shard");
+
+      final String finalFilePath = basePath + "/" + shardId;
+
+      final SequenceFile.Writer.Option[] writerOptions = new SequenceFile.Writer.Option[]{
+          SequenceFile.Writer.file(new Path(finalFilePath)),
+          SequenceFile.Writer.keyClass(ImmutableBytesWritable.class),
+          SequenceFile.Writer.valueClass(Result.class),
+          SequenceFile.Writer.compression(CompressionType.BLOCK)
+      };
+
+      final long start = System.nanoTime();
+      long localStart = System.nanoTime();
+
+      long reportedTotalDuration = 0;
+
+      try (BoundedSource.BoundedReader<Result> reader = source.createReader(c.getPipelineOptions())) {
+        boolean hasMore = reader.start();
+        readDuration.addValue(System.nanoTime() - localStart);
```

Comment:
`System.nanoTime() - localStart` is repeated often.  Does it make sense to encapsulate that?